### PR TITLE
Link to asynk module instead of async-nats

### PR DESCRIPTION
**Pull request by aditsachde**
I was confused about the difference between `async-nats` and `asynk` but eventually came across #192. Hopefully this will help clarify things for new users.

### CODE REVIEW

**Review by Jarema — CHANGES_REQUESTED** [review]
Let's just remove the second link.

**Comment by Jarema** [comment]
Yes, that can be confusing.

Thanks for the contribution!

File: src/lib.rs [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
 //! architectures.
 //!
-//! For async API refer to the [`async-nats`] crate.
+//! For async API refer to the [`asynk`] module.
 //!
 //! For more information see [https://nats.io/].
 //!
 //! [https://nats.io/]: https://nats.io/
-//! [`async-nats`]: https://docs.rs/async-nats
+//! [`async`]: crate::async
```
Comment (aditsachde) [inline src/lib.rs:27]: Rust doc does not render these lines as links, they used for defining where the links point to.

Comment (Jarema) [reply src/lib.rs:27]: My bad :).
The module though is named `asynk`.

Comment (aditsachde) [reply src/lib.rs]: Ah not sure how I missed that, but turns out linking to a module does not require a link declaration.

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**